frontend/src/scenes/feature-flags/featureFlagLogic.tsx

```diff
@@ -84,7 +85,19 @@ export const featureFlagLogic = kea<featureFlagLogicType>({
         featureFlag: [
             null as FeatureFlagType | null,
             {
-                setFeatureFlag: (_, { featureFlag }) => featureFlag,
+                setFeatureFlag: (_, { featureFlag }) => {
+                    const newFeatureFlag = clone(featureFlag)
+                    newFeatureFlag.filters.groups = newFeatureFlag.filters.groups.map((group) => {
+                        if (group.properties) {
+                            return {
+                                ...group,
+                                properties: convertPropertyGroupToProperties(group.properties),
+                            }
+                        }
+                        return group
+                    })
+                    return newFeatureFlag
```

Comment:
Is this the right place for this logic? When would we run `setFeatureFlag(almostARealFeatureFlagButRequiresSomeTweaking)`? Just asking, not being sure :).

However, if this is the right place, then why not embrace the functional way of doing things?

```ts
const newFeatureFlag = {
    ...featureFlag,
    filters: {
        ...featureFlag.filters,
        groups: /* etc */
    }
}
```

Reply:
I am not sure actually, I just need to make sure wherever we're setting a feature flag that if the groups in that feature flag come back as a property group, it needs to be converted to flat properties. And this should be updated now without the clone library 😄

Reply:
Hmmmm, I think feature flag filters are stored completely separately, and never touch the `Filter` object for queries, so backend can't parse, and won't deal with new property groups here.

Same in every other model really, that has a `filters` key, but no corresponding `Filter` object generation (which uses the PropertyMixin) . Ouch, this is a bit confusing.

Basically implies:

## No new properties

1. Feature Flags
2. Actions
3. ???

## New Properties

1. Entities
2. Insight filters
3. by extension, dashboard filters^
4. Experimentation filters
5. ???

Reply:
though @EDsCODE should confirm too, I might be missing something 😅

Reply:
is there a downside in just applying the conversion logic uniformly? Even if feature flag logic doesn't need to account for new filtering format, it might be less headache to just treat it the same as properties elsewhere.

Reply:
confused, what's the consensus on this? I vote for just applying the conversion logic since that's what the typing is yelling at me for